pandas/tests/scalar/timestamp/test_timezones.py

```diff
@@ -28,15 +28,17 @@ def test_tz_localize_pushes_out_of_bounds(self):
         pac = Timestamp.min.tz_localize('US/Pacific')
         assert pac.value > Timestamp.min.value
         pac.tz_convert('Asia/Tokyo')  # tz_convert doesn't change value
-        with pytest.raises(OutOfBoundsDatetime):
-            Timestamp.min.tz_localize('Asia/Tokyo')
+        with tm.assert_produces_warning(RuntimeWarning):
```

Comment:
I think there is a missing check in localize itself, IOW need to check if something is 0